src/workerRunner.ts

```diff
@@ -190,9 +227,18 @@ export class WorkerRunner extends EventEmitter {
     }
     if (dir === 'before')
       all.reverse();
-    for (const hook of all)
-      await fixturePool.resolveParametersAndRunHookOrTest(hook);
+    let error: Error | undefined;
+    for (const hook of all) {
+      try {
+        await fixturePool.resolveParametersAndRunHookOrTest(hook);
+      } catch (e) {
+        // Always run all the hooks, and capture the first error.
```

Comment:
But it no longer makes sense. Should we handle this as with fixtures via instantiating those hooks and tracking their states?